app.js

```diff
@@ -1235,6 +1257,17 @@ define(function(require) {
 			});
 
 			parent.on('click', '.unassigned-number', function(event) {
+				// reset select all button
+				var $selectAllBtn = parent.find('#unassigned_select_all'),
+					$selectAllLabel = parent.find('.unassigned-select-all-label'),
+					$selectAllIcon = parent.find('.unassigned-select-all-icon');
+
+				$selectAllIcon.removeClass('fa-square-o');
+				$selectAllIcon.addClass('fa-check-square-o');
+
+				$selectAllBtn.removeClass('active');
+				$selectAllLabel.text(self.i18n.active().select_all);
+
```

Comment:
Here you update the state of your `#unassigned_select_all` div to `select all` every time a checkbox is checked, even when not required.

Also, think of what should happen when someone selects all the numbers manually: it should change the state of your `#unassigned_select_all` div to `unselect all`.

I recommend you rewrite the whole logic of this event handler to take into account the fact that every time a checkbox is clicked (selected/unselected), you need to check if the state of your `#unassigned_select_all` div should be updated or not.

I created a dummy PBX and added three numbers to your `ricardo test` account so you can use it and test all the different use cases.

Reply:
You are right,

I have added support for different scenarios,

Done!